dashboard/src/main/resources/templates/component.ftl

```diff
@@ -110,8 +110,30 @@
     </#if>
 
     <h2>Dependencies</h2>
-    <pre class="dependency-tree">${dependencyTree}</pre>
-    
+
+    <#macro formatDependencyNode tree currentNode parent>
+      <#-- tree: ListMultimap<DependencyPath, DependencyPath>, currentNode: DependencyPath -->
+      <#if currentNode.size() gt 0>
+        <#-- root node does not have any leaf -->
+        <#if parent.size() gt 0>
+          <#assign label = 'parent: ' + parent.getLeaf() />
+        <#else>
+          <#assign label = 'root' />
+        </#if>
+        <p class="DEPENDENCY_TREE_NODE" title="${label}">${currentNode.getLeaf()}</p>
```

Comment:
I may have messed this up when I wrote some of this. It's weird that a node has a leaf. A node should either be or not be a leaf.

Reply:
Every DependencyPath instance has a leaf. (the middle node is stored in `path`)